### ISSUE TITLE
Terminated tests - Illegal nodeId

### ISSUE THREAD

**Issue by lancelote** [intellij-rust/intellij-rust#8351]
## Environment

* **IntelliJ Rust plugin version:** 0.4.163.4342-213-nightly
* **Rust toolchain version:** 1.59.0-nightly (5531927e8 2021-12-16) x86_64-apple-darwin
* **IDE name and version:** CLion 2022.1 EAP (CL-221.2973)
* **Operating system:** macOS 11.4
* **Macro expansion engine:** new
* **Name resolution engine:** new
* **Additional experimental features:** org.rust.cargo.evaluate.build.scripts, org.rust.macros.proc

## Problem description

When I run tests the plugin reports them as "Terminated", the test tree shows a permanent spinning progress weel, and the exception is raised

```
java.lang.Throwable: [Cargo Test]: Illegal nodeId: TestSuiteStartedEvent{name='0', id='0', parentId='0', locationUrl='cargo:test://0', running=true}
	at com.intellij.openapi.diagnostic.Logger.error(Logger.java:182)
	at com.intellij.execution.testframework.sm.runner.GeneralTestEventsProcessor.logProblem(GeneralTestEventsProcessor.java:372)
	at com.intellij.execution.testframework.sm.runner.GeneralTestEventsProcessor.logProblem(GeneralTestEventsProcessor.java:358)
	at com.intellij.execution.testframework.sm.runner.GeneralIdBasedToSMTRunnerEventsConvertor.validateAndGetNodeId(GeneralIdBasedToSMTRunnerEventsConvertor.java:382)
	at com.intellij.execution.testframework.sm.runner.GeneralIdBasedToSMTRunnerEventsConvertor.findNode(GeneralIdBasedToSMTRunnerEventsConvertor.java:388)
	at com.intellij.execution.testframework.sm.runner.GeneralIdBasedToSMTRunnerEventsConvertor.doStartNode(GeneralIdBasedToSMTRunnerEventsConvertor.java:121)
	at com.intellij.execution.testframework.sm.runner.GeneralIdBasedToSMTRunnerEventsConvertor.onSuiteStarted(GeneralIdBasedToSMTRunnerEventsConvertor.java:114)
	at com.intellij.execution.testframework.sm.runner.OutputToGeneralTestEventsConverter.fireOnSuiteStarted(OutputToGeneralTestEventsConverter.java:285)
	at com.intellij.execution.testframework.sm.runner.OutputToGeneralTestEventsConverter$MyServiceMessageVisitor.visitTestSuiteStarted(OutputToGeneralTestEventsConverter.java:384)
	at jetbrains.buildServer.messages.serviceMessages.TestSuiteStarted.visit(TestSuiteStarted.java:32)
	at com.intellij.execution.testframework.sm.runner.OutputToGeneralTestEventsConverter.processServiceMessage(OutputToGeneralTestEventsConverter.java:135)
	at com.intellij.execution.testframework.sm.runner.OutputToGeneralTestEventsConverter.processServiceMessages(OutputToGeneralTestEventsConverter.java:129)
	at org.rust.cargo.runconfig.test.CargoTestEventsConverter.handleSuiteMessage(CargoTestEventsConverter.kt:132)
	at org.rust.cargo.runconfig.test.CargoTestEventsConverter.processServiceMessages(CargoTestEventsConverter.kt:57)
	at com.intellij.execution.testframework.sm.runner.OutputToGeneralTestEventsConverter.processConsistentText(OutputToGeneralTestEventsConverter.java:113)
	at com.intellij.execution.testframework.sm.runner.OutputToGeneralTestEventsConverter$1.onTextAvailable(OutputToGeneralTestEventsConverter.java:69)
	at com.intellij.execution.testframework.sm.runner.OutputEventSplitter.onTextAvailableInternal(OutputEventSplitter.kt:79)
	at com.intellij.execution.testframework.sm.runner.OutputEventSplitter.onTextAvailable(OutputEventSplitter.kt:74)
	at com.intellij.smRunner.OutputEventSplitterBase.flushInternal(OutputEventSplitterBase.kt:159)
	at com.intellij.smRunner.OutputEventSplitterBase.flushInternal$default(OutputEventSplitterBase.kt:145)
	at com.intellij.smRunner.OutputEventSplitterBase.processInternal(OutputEventSplitterBase.kt:93)
	at com.intellij.smRunner.OutputEventSplitterBase.process(OutputEventSplitterBase.kt:74)
	at com.intellij.execution.testframework.sm.runner.OutputEventSplitter.process(OutputEventSplitter.kt:59)
	at com.intellij.execution.testframework.sm.runner.OutputToGeneralTestEventsConverter.process(OutputToGeneralTestEventsConverter.java:95)
	at com.intellij.execution.testframework.sm.SMTestRunnerConnectionUtil$2.onTextAvailable(SMTestRunnerConnectionUtil.java:218)
	at jdk.internal.reflect.GeneratedMethodAccessor16.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at com.intellij.execution.process.ProcessHandler$2.invoke(ProcessHandler.java:247)
	at com.sun.proxy.$Proxy34.onTextAvailable(Unknown Source)
	at com.intellij.execution.process.ProcessHandler.notifyTextAvailable(ProcessHandler.java:221)
	at org.rust.cargo.runconfig.RsProcessHandler.coloredTextAvailable(RsProcessHandler.kt:31)
	at com.intellij.execution.process.AnsiEscapeDecoder.processTextChunk(AnsiEscapeDecoder.java:100)
	at com.intellij.execution.process.AnsiEscapeDecoder.escapeText(AnsiEscapeDecoder.java:64)
	at org.rust.cargo.runconfig.RsAnsiEscapeDecoder.escapeText(RsAnsiEscapeDecoder.kt:36)
	at org.rust.cargo.runconfig.RsProcessHandler.notifyTextAvailable(RsProcessHandler.kt:27)
	at com.intellij.execution.process.BaseOSProcessHandler$SimpleOutputReader.onTextAvailable(BaseOSProcessHandler.java:155)
	at com.intellij.util.io.BaseOutputReader.sendText(BaseOutputReader.java:205)
	at com.intellij.util.io.BaseOutputReader.processInput(BaseOutputReader.java:189)
	at com.intellij.util.io.BaseOutputReader.readAvailableNonBlocking(BaseOutputReader.java:101)
	at com.intellij.util.io.BaseDataReader.readAvailable(BaseDataReader.java:74)
	at com.intellij.util.io.BaseDataReader.doRun(BaseDataReader.java:155)
	at com.intellij.util.ConcurrencyUtil.runUnderThreadName(ConcurrencyUtil.java:212)
	at com.intellij.util.io.BaseDataReader.lambda$start$0(BaseDataReader.java:50)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:668)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1$1.run(Executors.java:665)
	at java.base/java.security.AccessController.doPrivileged(Native Method)
	at java.base/java.util.concurrent.Executors$PrivilegedThreadFactory$1.run(Executors.java:665)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

<img width="959" alt="tree" src="https://user-images.githubusercontent.com/2500320/147853518-8015fb43-37cc-49f6-94c6-ae864c26ebc6.png">

## Steps to reproduce

- Create a new library project
- Run default test
